### PR TITLE
fix(Scripts/ShadowmoonValley): Rewrite Dragonmaw Races

### PR DESCRIPTION
**Pull request by Gultask**
This is an improvement, but not a complete fix. It should use a lot of spells instead.

## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/6943

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

Mostly from 🥭s

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. mod the rep
2. run the query below
```
UPDATE `quest_template_addon` SET `PrevQuestID` = 0 WHERE (`ID` IN (11064,11067,11068,11069,11070,11071));
```
3. go to the quest area, it works with .gm on

## Known Issues and TODO List:

- [x] Summons only appear on the ground
- [x] Cannot seem to make them actually *fly*

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
```
/*
UPDATE `quest_template_addon` SET `PrevQuestID` = 0 WHERE (`ID` IN (11064,11067,11068,11069,11070,11071));

UPDATE `quest_template_addon` SET `PrevQuestID` = 11063 WHERE `ID`=11064;
UPDATE `quest_template_addon` SET `PrevQuestID` = 11064 WHERE `ID`=11067;
UPDATE `quest_template_addon` SET `PrevQuestID` = 11067 WHERE `ID`=11068;
UPDATE `quest_template_addon` SET `PrevQuestID` = 11068 WHERE `ID`=11069;
UPDATE `quest_template_addon` SET `PrevQuestID` = 11069 WHERE `ID`=11070;
UPDATE `quest_template_addon` SET `PrevQuestID` = 11070 WHERE `ID`=11071;
*/
```

**Comment by Gultask** [comment]
@elthehablo check this out :3

**Comment by Gultask** [comment]
I was being careful to avoid nullptr crashes but y'know.

**Comment by elthehablo** [comment]
He did it!

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 1]
```diff
@@ -1830,17 +1830,423 @@ class spell_disrupt_summoning_ritual : public SpellScript
...
+    SAY_START               = 0,
+    SAY_COMPLETE            = 1,
+    SAY_SKYSHATTER_SPECIAL  = 2,
+};
+
+struct dragonmaw_race_npc : public ScriptedAI
+{
+    dragonmaw_race_npc(Creature* creature) : ScriptedAI(creature) { }
+
+    Player* _player;
```
Comment (elthehablo) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1878]: This is indicated as a private variable, so best to define it in the private section

Comment (Gultask) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp]: like this?
image

Comment (elthehablo) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp]: Yup :D

**Comment by Gultask** [comment]
Wanted someone to test it :3

Otherwise I also did it myself and it's completable.

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 2]
```diff
@@ -3609,62 +3609,6 @@ void SpellMgr::LoadSpellInfoCorrections()
...
-    });
-
-    // Sky Shatter
-    ApplySpellFix({ 40945 }, [](SpellInfo* spellInfo)
-    {
-        spellInfo->Targets |= TARGET_FLAG_DEST_LOCATION;
-        spellInfo->Effects[EFFECT_0].TriggerSpell = 41064;
-        spellInfo->Effects[EFFECT_0].Effect = SPELL_EFFECT_TRIGGER_MISSILE;
-        spellInfo->Effects[EFFECT_0].TargetA = SpellImplicitTargetInfo(TARGET_DEST_DEST);
-    });
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellInfoCorrections.cpp:3666]: beautiful

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 3]
```diff
@@ -1830,17 +1830,427 @@ class spell_disrupt_summoning_ritual : public SpellScript
...
+    void StartRace()
+    {
+        me->SetWalk(false);
+        ScheduleTimedEvent(5s, [&]
+        {
+                if (!_player)
+                    FailQuest();
+                else if (!me->IsWithinDist(_player, 100.f))
+                    FailQuest();
+        }, 5s);
```
Comment (Kitzunu) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1936]: indentation here is fucked

File: src/server/scripts/Spells/spell_quest.cpp [hunk 4]
```diff
@@ -437,68 +437,24 @@ enum q11520Roots
...
-        Unit* caster = GetCaster();
-        if (Creature* trigger = caster->SummonCreature(23356, _x, _y, _z, 0.0f, TEMPSUMMON_TIMED_DESPAWN, 1500))
+        if (GameObject* go = GetCaster()->FindNearestGameObject(GO_RAZORTHORN_DIRT_MOUNT, 20.0f))
         {
-            trigger->CastSpell(trigger, GetSpellInfo()->Effects[effIndex].TriggerSpell, true);
-            if (GetSpellInfo()->Effects[effIndex].TriggerSpell == 41064)
-                trigger->CastSpell(trigger, 41284, true);
+            GetCaster()->GetMotionMaster()->MovePoint(0, *go);
+            go->SetLootState(GO_JUST_DEACTIVATED);
+            GetCaster()->CastSpell(GetCaster(), SPELL_SUMMON_RAZORTHORN_ROOT, true);
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_quest.cpp:450]: You need to validate spells used in spellscripts

**Comment by Kitzunu** [comment]
and conflict

File: src/server/scripts/Spells/spell_quest.cpp (hunk 4 above)
Comment (Gultask) [reply src/server/scripts/Spells/spell_quest.cpp:450]: that one isn't mine, I just removed the script that was used for dragonmaw races :P

**Comment by Gultask** [comment]
do I think

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**